### ISSUE TITLE
configure_optimizers passes frozen backbone params to AdamW, wasting optimizer memory

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#23]
### Summary
`configure_optimizers()` in `src/models/multi_head_resnet.py` passes `self.parameters()` to AdamW, which includes all backbone parameters — even when the backbone is frozen (`requires_grad=False`). While PyTorch skips gradient updates for frozen parameters, it still allocates optimizer state (momentum buffer + variance estimate) for them, roughly doubling the memory footprint of the backbone weights during the frozen warm-up phase.

For ResNet18, the backbone has ~11M parameters (~44 MB in float32). The optimizer state adds another ~88 MB unnecessarily during freeze epochs.

### Code location
`src/models/multi_head_resnet.py`, `configure_optimizers()`, line 296:

```python
optimizer = torch.optim.AdamW(
    self.parameters(),   # includes frozen backbone params
    ...
)
```

### Proposed change
Pass only the trainable parameters to the optimizer. When the backbone is later unfrozen, Lightning handles re-initialisation automatically if `configure_optimizers` is called via `LightningModule.optimizers()` after `on_train_epoch_start`.

A simpler approach that avoids optimizer re-init complexity is to use parameter groups:

```python
optimizer = torch.optim.AdamW([
    {'params': self.heads.parameters()},
    {'params': self.backbone.parameters(), 'lr': self.hparams.lr / 10},
], lr=self.hparams.lr, weight_decay=self.hparams.weight_decay)
```

Or just filter at construction time and accept that frozen params get state allocated (current behaviour but documented as a known tradeoff).

### Files affected
- `src/models/multi_head_resnet.py`
